scipy/optimize/minpack.py

```diff
@@ -671,6 +672,16 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     >>> plt.show()
 
     """
+
+    def check_len(z):
+        if len(z) == 0:
+            return False
+        else:
+            return True
+
+    if check_len(xdata) is False or check_len(ydata) is False:
+        raise TypeError("Improper input: N=1 must not exceed M=0")
```

Comment:
If it isn't too much trouble, may be good to add the example from #9864 as a unit test?

Reply:
```
from scipy.optimize import curve_fit
curve_fit(lambda x, a: a*x, [], [])
TypeError: Improper input: N=1 must not exceed M=0 
```
```
from scipy.optimize import curve_fit
curve_fit(lambda x, a: a*x, [], [], bounds = (0, 1))
TypeError                                 Traceback (most recent call last)
<ipython-input-7-556b79738194> in <module>()
      1 from scipy.optimize import curve_fit
----> 2 curve_fit(lambda x, a: a*x, [], [], bounds = (0, 1))

~\Anaconda3\lib\site-packages\scipy\optimize\minpack.py in curve_fit(f, xdata, ydata, p0, sigma, absolute_sigma, check_finite, bounds, method, jac, **kwargs)
    682 
    683     if check_len(xdata) is False:
--> 684         raise TypeError("Improper input: N=1 must not exceed M=0")
    685 
    686     if p0 is None:

TypeError: Improper input: N=1 must not exceed M=0
```
I was not sure where to add the example so I added it here, in both the cases with `bounds` and without `bounds` it shows same error message

Reply:
I added a test case here https://github.com/scipy/scipy/pull/9903

Reply:
Can you add it here ?

Reply:
I added it to a file named test_minpack.py, you can find that here https://github.com/scipy/scipy/tree/master/scipy/optimize/tests

Reply:
I mean can you add it to this PR if you don't mind so we can check it together in one place?

Reply:
I did just now.